src/main/java/imsam/SpecifiedDegreeDistribution.java

```diff
@@ -31,7 +33,7 @@ public Pair(T f, U s) {
 	 *
 	 * @param itemProbabilityPairs Items and probabilities they are chosen
 	 * */
-	public SpecifiedDegreeDistribution(ArrayList<Pair<T, Float>> itemProbabilityPairs) {
+	public SpecifiedDegreeDistribution(ArrayList<Pair<T, Float>> itemProbabilityPairs) throws Exception {
```

Comment:
Not sure if the constructor of a class should throw an Exception. If so, it should be a specific type of Exception, i.e., you could create one called `InvalidProbabilityPairsException`